src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -663,6 +663,11 @@ namespace cryptonote
     bl.minor_version = CURRENT_BLOCK_MINOR_VERSION;
     bl.timestamp = 0;
     bl.nonce = nonce;
+
+    #if defined __EMSCRIPTEN__  // miner not accessible in webassembly
+      return false;
+    #endif
```

Comment:
If you can't generate the genesis block, you can't even start the wallet2 hash chain. There is a C implementation of Cryptonight. Does it not work ?

Reply:
The wallet2 instances created through webassembly generally work; they're able to get the balance, send, receive, etc.  Is there a problem I can expect to see?

Can you elaborate how to use the C implementation of Cryptonight here?

Reply:
They'll get a wrong genesis block hash. I guess it might not matter in practice for the wallet only...

To use the C version of Cryptonight, you ensure the C version is compiled in slow-hash.c. It should be the case if you define NO_AES, looking at the code.

Reply:
That worked.  I've removed this change.